opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/SpanBuilderShim.java

```diff
@@ -232,4 +254,69 @@ public Span start() {
 
     return new SpanShim(telemetryInfo(), span, baggage);
   }
+
+  // The first SpanContext with Child Of type in the entire list is used as parent,
+  // else the the first SpanContext is used as parent.
+  @Nullable
+  static io.opentelemetry.api.trace.SpanContext getMainParent(List<SpanParentInfo> parents) {
+    if (parents.size() == 0) {
+      return null;
+    }
+
+    SpanParentInfo mainParent = null;
+
+    for (SpanParentInfo parentInfo : parents) {
+      if (parentInfo.getRefType() == ReferenceType.CHILD_OF) {
+        mainParent = parentInfo;
+        break;
+      }
+    }
+
+    mainParent = mainParent == null ? parents.get(0) : mainParent;
+    return mainParent.getSpanContext();
```

Comment:
Minor simplification suggestion:
```suggestion
    SpanParentInfo mainParent = parents.get(0);
    for (SpanParentInfo parentInfo : parents) {
      if (parentInfo.getRefType() == ReferenceType.CHILD_OF) {
        mainParent = parentInfo;
        break;
      }
    }
    return mainParent.getSpanContext();
```

Reply:
Updated.